common/functions/src/aggregates/aggregate_retention.rs

```diff
@@ -142,18 +141,17 @@ impl AggregateFunction for AggregateRetentionFunction {
         array: &mut dyn common_datavalues::MutableColumn,
     ) -> Result<()> {
         let state = place.get::<AggregateRetentionState>();
-        let builder: &mut MutableObjectColumn<VariantValue> =
-            Series::check_get_mutable_column(array)?;
-        let mut vec: Vec<u8> = vec![0; self.events_size as usize];
+        let builder: &mut MutableArrayColumn = Series::check_get_mutable_column(array)?;
+        let mut vec: Vec<DataValue> = vec![0u8.into(); self.events_size as usize];
         if state.events & 1 == 1 {
-            vec[0] = 1;
+            vec[0] = 1u8.into();
             for i in 1..self.events_size {
                 if state.events & (1 << i) != 0 {
-                    vec[i as usize] = 1;
+                    vec[i as usize] = 1u8.into();
                 }
             }
         }
-        builder.append_value(VariantValue::from(json!(vec)));
+        builder.append_value(ArrayValue::new(vec.clone()));
```

Comment:
It's incorrect.

We should build UInt8Column & Int64Offsets.

Reply:
oh, yes.

Reply:
yes, build an inner uint8 column and offsets, use function `ArrayColumn::from_data` to build `ArrayColumn`

Reply:
ok, thank you very much

Reply:
make sense. thank you.